test/suite/test_cursor_bound_fuzz.py

```diff
@@ -108,129 +118,262 @@ def dump_key_range(self):
         for i in self.key_range_iter():
             self.pr(self.key_range[i].to_string())
 
+    # Generate a random ascii value.
     def generate_value(self):
         return ''.join(random.choice(string.ascii_lowercase) for _ in range(self.value_size))
 
+    # Get a value from the value array.
     def get_value(self):
         return self.value_array[random.randrange(self.value_array_size)]
 
-    def apply_update(self, cursor, key_id):
+    # Get a key within the range of min_key and max_key.
+    def get_random_key(self):
+        return random.randrange(self.min_key, self.max_key)
+
+    # Update a key using the cursor and update its in memory representation.
+    def apply_update(self, cursor, key_id, prepare):
         value = self.get_value()
         cursor[key_id] = value
-        self.key_range[key_id].update(value, key_states.UPSERTED)
-        self.verbose(3, "Updating key: " + str(key_id))
+        self.key_range[key_id].update(value, key_states.UPSERTED, self.current_ts, prepare)
+        self.verbose(3, "Updating " + self.key_range[key_id].to_string())
 
-    def apply_remove(self, cursor, key_id):
+    # Remove a key using the cursor and mark it as deleted in memory.
+    # If the key is already deleted we skip the remove.
+    def apply_remove(self, cursor, key_id, prepare):
         if (self.key_range[key_id].is_deleted()):
             return
-        self.verbose(3, "Removing key: " + str(key_id))
         cursor.set_key(key_id)
         self.assertEqual(cursor.remove(), 0)
-        self.key_range[key_id].update(None, key_states.DELETED)
-
-    def apply_ops(self, cursor):
+        self.key_range[key_id].update(None, key_states.DELETED, self.current_ts, prepare)
+        self.verbose(3, "Removing " + self.key_range[key_id].to_string())
+
+    # Apply a truncate operation to the key range.
+    def apply_truncate(self, session, cursor, cursor2, prepare):
+        lower_key = self.get_random_key()
+        if (lower_key + 1 < self.max_key):
+            upper_key = random.randrange(lower_key + 1, self.max_key)
+            cursor.set_key(lower_key)
+            cursor2.set_key(upper_key)
+            self.assertEqual(session.truncate(None, cursor, cursor2, None), 0)
+
+            for key_id in range(lower_key, upper_key + 1):
```

Comment:
np: I would note here that we do a +1 because of the range inclusivity rules python has